### PR TITLE
C4/journey card menu

### PR DESCRIPTION
**Pull request by csiyang**

### CODE REVIEW

File: apps/journeys-admin/src/components/JourneyList/journeyListData.ts [hunk 1]
```diff
@@ -9,5 +13,6 @@ export const defaultJourney: Journey = {
   themeName: ThemeName.base,
   themeMode: ThemeMode.light,
   slug: 'default',
-  locale: 'en_US'
+  locale: 'en_US',
+  status: JourneyStatus.published
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/journeyListData.ts:17]: In another PR we made this draft

```suggestion
  status: JourneyStatus.draft
```

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx [hunk 2]
```diff
@@ -1,26 +1,75 @@
...
+      />
+    )
+
+    expect(getByRole('button')).toHaveAttribute(
+      'aria-controls',
+      'journey-actions'
+    )
+    expect(getByRole('button')).toHaveAttribute('aria-haspopup', 'true')
+    expect(getByRole('button')).toHaveAttribute('aria-expanded', 'false')
+    fireEvent.click(getByRole('button'))
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx:20]: Should also check     `expect(getByRole('button')).toHaveAttribute('aria-expanded', 'true')` after button is clicked

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx [hunk 3]
```diff
@@ -1,26 +1,75 @@
...
+    expect(getByRole('menu')).toHaveAttribute(
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx:21]: Check this before clicking the button? It's something that should be set no matter if you click the button or not.

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx [hunk 4]
```diff
@@ -1,16 +1,27 @@
 import { Story, Meta } from '@storybook/react'
-
 import { journeysAdminConfig } from '../../../../libs/storybook'
 import JourneyCardMenu from './JourneyCardMenu'
+import { JourneyStatus } from '../../../../../__generated__/globalTypes'
 
-const TestStory = {
+const JoruneyCardMenuDemo = {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx:6]:
```suggestion
const JourneyCardMenuDemo = {
```
Typo is repeated a couple of times down this page as well

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
   title: 'Journeys-Admin/JourneyList/JourneyCard/JourneyCardMenu'
 }
 
-const Template: Story = () => <JourneyCardMenu />
+const Template: Story = ({ ...args }) => (
+  <JourneyCardMenu status={args.status} slug={args.slug} />
+)
 
-export const StoryComponent = Template.bind({})
+export const DraftJoruneyCardMenu = Template.bind({})
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx:16]:
```suggestion
export const Draft = Template.bind({})
```

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
+DraftJoruneyCardMenu.args = {
+  status: JourneyStatus.draft,
+  slug: 'draft-journey'
+}
+export const PublishedJoruneyCardMenu = Template.bind({})
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx:21]:
```suggestion
export const Published = Template.bind({})
```

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.tsx [hunk 7]
```diff
@@ -48,10 +53,43 @@ const JourneyCardMenu = (): ReactElement => {
...
+          <MenuItem>
+            <ListItemIcon>
+              <EditIcon />
+            </ListItemIcon>
+            <ListItemText>Edit</ListItemText>
+          </MenuItem>
+        </Link>
+
+        {/* update link */}
+        <Link href={`/journeys/${slug}/access`} underline="none">
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.tsx:66]: nope, just a placeholder, hence the comment above it

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx (hunk 3 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx:21]: menu does not appear in the DOM before the button is clicked on

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx [hunk 8]
```diff
@@ -1,16 +1,27 @@
...
 
-const TestStory = {
+const JoruneyCardMenuDemo = {
   ...journeysAdminConfig,
   component: JourneyCardMenu,
   title: 'Journeys-Admin/JourneyList/JourneyCard/JourneyCardMenu'
 }
 
-const Template: Story = () => <JourneyCardMenu />
+const Template: Story = ({ ...args }) => (
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx:12]: How do we force this to open? Would I need to update the component to allow for this?

[the remark this answers is not in the document]
Comment (storyworks) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.stories.tsx:12]: Yeah you would.. add the open prop to JourneyCardMenuProps - you can make it an optional `open?: boolean`

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx (hunk 3 above)
Comment (storyworks) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/JourneyCardMenu.spec.tsx:21]: Good point!

**Pull request merged**